File: boj/BinarySearch/BOJ_1072.cpp
// 백준 1072 게임
// 21.02.17
#include <iostream>
#include <algorithm>
#include <queue>
#include <vector>
#include <string>
#include <cmath>
#include <set>
#define INF 2147000000

typedef long long ll;

using namespace std;

int main()
{
    ll x, y; scanf("%lld %lld", &x, &y);
    ll z = (100 * y) / x;
    int l = 0;
    int r = 1000000000;
    int m = (l + r) / 2;
    int ans = INF;

    while (l <= r)
    {
        ll temp = (100 * (y + m)) / (x + m);

        if (temp > z)
        {
            ans = min(ans, m);
            r = m - 1;
        }
        else
        {
            l = m + 1;
        }

        m = (l + r) / 2;
    }
    if (ans < INF) printf("%d\n", ans);
    else printf("%d\n", -1);

    return 0;
}

File: boj/BinarySearch/BOJ_1072.swift
//  백준 1072 게임
//  21.02.17

import Foundation
let INF: Int = 2147000000

let input = readLine()!.split(separator: " ").map { Int(String($0))! }
let x = input[0]
let y = input[1]
let z = (100 * y) / x

var l = 0
var r = 1000000000
var m = (l + r) / 2
var ans = INF

if z >= 99 { print(-1) }
else {
  while l <= r {
  let temp = (100 * (y + m)) / (x + m)
    
    if temp > z {
      ans = min(ans, m)
      r = m - 1
    } else {
      l = m + 1
    }
    m = (l + r) / 2
  }
  print(ans)
}
